Release v5.0.0

gmaclennan opened the pull request:
This PR is to prepare for release v5.0.0 (previously v2.1.0, but bumping the version number to bring in-line with Mapeo Desktop). This is testing a new release process, where "Release Candidates" for testing will be shared in this PR. When it has passed QA testing then this will be merged with the `deploy` branch and released to Google Play, Github Releases, etc.

See the [draft changelog](https://github.com/digidem/mapeo-mobile/blob/release/v2.1.0/CHANGELOG.md) for changes since our last public release (v2.0.0-beta.5)

Note that previously we released "betas" publicly. From now on we no longer use "beta" and "alpha" terms.

**APKs for testing**
- [QA v5.0.0-RC.844](https://mapeo-apks.s3.amazonaws.com/beta-testing/v5.0.0/mapeo-v5.0.0-RC.844-QA-debug.apk): You can install this alongside your existing Mapeo App and test out new features without affecting your existing data. Look for the red logo.
- ⚠️[v5.0.0-RC.844 APK](https://mapeo-apks.s3.amazonaws.com/beta-testing/v5.0.0/mapeo-v5.0.0-RC.844-universal.apk): This will replace your existing Mapeo installation and you will not be able to downgrade - if it does not work you will need to wait for a fix.

gmaclennan committed:
fix: Adjust projectKey styling and text

gmaclennan committed:
fix(Sync): Touch up styles of no wifi sync screen

gmaclennan committed:
chore(docs): Add README note about breaking changes and stability

gmaclennan committed:
chore(docs): Add README note about translations

gmaclennan committed:
feat: Update to latest React Native & Unimodules (better performance and bug fixes)

Squashed commits:
[2cdd3e6] chore: Cleanup unused vars (+7 squashed commits)
[2599762] chore: Fix gradlew script (was throwing warning)
[dd60087] chore: Update deps
[c41f02d] fix: Update react-native-v8 for RN 0.61
[556422c] Update to latest unimodules
[737c490] remove redundant patch (issue was fixed in RN61)
[8cf62f0] Bump react-native-fs to latest
[37a7e1e] Update to RN 0.61.5

gmaclennan committed:
fix: Don't render app until saved draft is loaded from storage

gmaclennan committed:
fix: Fix crash after deleting an observation's description

gmaclennan committed:
feat: Use Android default animation for navigation (changes close icon on GPS and Sync screens)

Simplify the navigation structure to remove nested stacks and use the built-in react-navigation headers.
Change the way navigation is blocked for user confirmation after creating or editing an observation.
Uses back arrow (not X close icon) on GPS and Sync modals, since this is the Android default.

gmaclennan committed:
feat: Ask for confirmation when the user presses the back button when editing/creating an observation

gmaclennan committed:
fix: Fix bug when a user edits an existing observation and is returned to the home screen (map)

gmaclennan committed:
chore(release): 2.0.0-alpha.0

gmaclennan committed:
feat: 🎉 Import configuration files (.mapeosettings) from a new settings screen

gmaclennan committed:
chore(release): 2.1.0-alpha.1

gmaclennan committed:
fix: Remove unused picker component (pending coordinate format setting)

gmaclennan committed:
chore(release): 2.1.0-alpha.2

gmaclennan committed:
fix: Fix config import failing when there is an existing config

gmaclennan committed:
chore(release): 2.1.0-alpha.3

gmaclennan committed:
New Crowdin translations (#272)

* New translations en.json (Burmese)

* New translations en.json (Khmer)

* New translations en.json (Lao)

* New translations en.json (Thai)

* New translations en.json (Vietnamese)

* New translations en.json (Hindi)

* New translations en.json (Indonesian)

* New translations en.json (Malay)

* New translations en.json (Nepali)

* New translations en.json (Nepali)

* New translations en.json (Nepali)

* New translations en.json (Nepali)

* New translations en.json (Nepali)

* New translations en.json (Nepali)

* New translations en.json (Indonesian)

* New translations en.json (Nepali)

* New translations en.json (Nepali)

* New translations en.json (Nepali)

* New translations en.json (Nepali)

* New translations en.json (Spanish)

* New translations en.json (French)

* New translations en.json (Spanish)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Burmese)

* New translations en.json (Khmer)

* New translations en.json (Lao)

* New translations en.json (Thai)

* New translations en.json (Vietnamese)

* New translations en.json (Hindi)

* New translations en.json (Indonesian)

* New translations en.json (Malay)

* New translations en.json (Nepali)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Burmese)

* New translations en.json (French)

* New translations en.json (Hindi)

* New translations en.json (Indonesian)

* New translations en.json (Khmer)

* New translations en.json (Lao)

* New translations en.json (Malay)

* New translations en.json (Nepali)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Spanish)

* New translations en.json (Thai)

* New translations en.json (Vietnamese)

gmaclennan committed:
fix: Fix crash when presets have empty placeholder field

gmaclennan committed:
feat: Add settings screen to change language (independent of device language) (#302)

* Abstract persisted state to a custom hook

* Add context for user-selected locale

* Change to list-based settings screen

* Add language settings screen

* move settings icon onto observationList screen

* Extract translatable strings

* Only extract `en` messages - crowdin will deal with the rest

* Make language screen scrollable

* Sort languages by english name

* Include English name in language list

* add dense list settings for formatting settings selectone field

* fix: add react-native-screens

gmaclennan committed:
chore(release): 2.1.0-alpha.4

okdistribute committed:
karissa->okdistribute

okdistribute committed:
fix: add package-lock.json files for front and backend

okdistribute committed:
Merge pull request #308 from digidem/package-lock

fix: add package-lock.json files for front and backend

gmaclennan committed:
chore: Enable package-lock in .npmrc

gmaclennan committed:
fix: Fix crash caused by react-native-v8 patch release fixes #311

gmaclennan committed:
chore: Report "Application Not Responding" errors to Bugsnag

gmaclennan committed:
chore: Add bugsnag gradle / Android integration

gmaclennan committed:
feat: Release smaller APK for Arm devices only (#322)

Almost all phones use Arm chips, so this reduces the size by about 50% with little impact. Users of Intel devices (some rare phones from about 2015 and some Chromebooks that run Android apps) will need to use the Intel or Universal APKs.

* Do not include support x86 in the main release APK

* Add intel-only build for x86 phones

Very few Android devices, apart from Chromebooks, are x86.
This would mainly be for the 2015 ASUS Zenphone in case users want to use it.
The PlayStore release will continue to be cross-platform, but splitting the APK into two releases reduces the APK size by about 40%.

* Add universal build

* More sensible naming of output apks

* Only create universal and debug build for QA

* remove unused dep

This was only used for yarn with patch-package, and we don't use yarn.

* Fix for running `npm ci` on CI

* Add bitrise.yml to repo

* Remove exposed AWS key (exposed key de-activated)

* Change keystore passwords and move to env variables

* Setup bitrise.yml to run from repo

* Sign QA builds with a separate key

* Adjust apk naming

* Setup beta and release workflows

* rename workflows

dependabot[bot] committed:
chore(deps): bump acorn from 7.1.0 to 7.1.1 in /src/backend

Bumps [acorn](https://github.com/acornjs/acorn) from 7.1.0 to 7.1.1.
- [Release notes](https://github.com/acornjs/acorn/releases)
- [Commits](https://github.com/acornjs/acorn/compare/7.1.0...7.1.1)

Signed-off-by: dependabot[bot] <support@github.com>

gmaclennan committed:
feat: New translations including Vietnamese (#303)

Squashed commits:
[d950853] New translations en.json (Vietnamese)
[9496dac] New translations en.json (Vietnamese)
[f1ce80e] New translations en.json (Vietnamese)
[aa182a3] New translations en.json (Vietnamese)
[415ed78] New translations en.json (Vietnamese)
[6b2261a] New translations en.json (Vietnamese)
[7743cae] New translations en.json (Thai)
[07db0e2] New translations en.json (Spanish)
[c7dee5c] New translations en.json (Portuguese, Brazilian)
[0adf598] New translations en.json (Nepali)
[d3d753c] New translations en.json (Malay)
[ab64667] New translations en.json (Lao)
[05abda9] New translations en.json (Khmer)
[35bf81b] New translations en.json (Indonesian)
[35998db] New translations en.json (Hindi)
[4095b4f] New translations en.json (French)
[4faf387] New translations en.json (Burmese)

gmaclennan committed:
chore: small improvements to logging for nodejs-mobile

gmaclennan committed:
feat: New "Mapeo for ICCAs" variant for UNEP WCMC (#346)

* feat: Add intro screens for ICCA / WCMC

* feat: Add icca presets for icca app flavor

* feat: Only show icca intro screens for icca build flavor

* feat: Add ICCA splash screen and logo

* chore: package-lock updates

* Build ICCA for beta

* fix package-lock

* remove unused dep

* add signing config for mapeo for iccas

* fix S3 upload and ignore failed S3 uploads in the future

* Include ICCA variant in S3 release & beta uploads

* Update download URLs to point to latest release

* Fix syntax error in APK path names

* [skip ci] fix script location

* Fix path for google play upload

* debug source_dir

* Fix - change working dir after gradle build

* Another fix for scripts :/

gmaclennan committed:
fix: Bundle React-Native code in Universal & Intel build types

gmaclennan committed:
chore: Improve deploy process (#349)

* Update download URLs when releasing

https://apk.mapeo.world/latest
https://apk.mapeo.world/icca/latest

* Remove Intel-only builds (Universal build works for this)

* Update apk naming

The makes the names easier to parse when in a long list

gmaclennan committed:
chore: Update deps (including flow)

gmaclennan committed:
chore: remove unused yarn artifacts

gmaclennan committed:
chore: Switch out deprecated react-native-hooks package

gmaclennan committed:
Fix: Fix crash from switching away from app during camera view

We've seen several errors in bugsnag caused by subscriptions and the expo-sensors module.
Turns out we were doing too much, trying to unsubscribe from the accelerometer when the app was going into the background. Expo was already doing that, so we were throwing an error by doing it twice. Re-writing this as a function component helped clear up the logic and identify the bugs.
fixes #330 #187 #138

gmaclennan committed:
fix: Don't [silently] throw error if a user cancels a share action

fixes #344

gmaclennan committed:
Fix: Fix double header on photo modal (from alphas)

gmaclennan committed:
FIx: Fix race conditions with server start-stop (switching away from app and back)

Fixes #210 (hopefully!)

gmaclennan committed:
chore: Improve debug messages and error reporting

gmaclennan committed:
fix: Fix online mapstyle not loading when offline style is removed

gmaclennan committed:
fix: Do not update user location (blue dot) on map until user moves more than 15 meters (saves battery life)

This reduces CPU use on a Moto 4 from 150% to 20% when the map is not following the user
TODO: Reduce updates to map when in follow mode to reduce CPU usage further

gmaclennan committed:
feat: Show "empty state" when user has not yet recorded any observations

gmaclennan committed:
chore: Fix gradle build errors trying to strip mips code

gmaclennan committed:
fix: Attempt to fix builds for OPPO phones (fixes #301)

gmaclennan committed:
fix: Fix translations build script

gmaclennan committed:
chore: Update translations for new features (#350)

* New translations en.json (Spanish)

* New translations en.json (Spanish)

* New translations en.json (Spanish)

* New translations en.json (Spanish)

* New translations en.json (Spanish)

* New translations en.json (Spanish)

* New translations en.json (Vietnamese)

* New translations en.json (Burmese)

* New translations en.json (French)

* New translations en.json (Hindi)

* New translations en.json (Indonesian)

* New translations en.json (Khmer)

* New translations en.json (Lao)

* New translations en.json (Malay)

* New translations en.json (Nepali)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Thai)

* New translations en.json (Spanish)

gmaclennan committed:
feat: Add end-to-end tests with Detox (#355)

Fixes #310

* chore: Move home screen header and add button into parent components

* feat: Add detox and initial test

* feat: Add github action for detox tests

* Add cache

* Switch to custom emulator setup

* checkout first

* Use npm ci instead of npm install

* Fix curl script

* move patch-package to non-dev dep

* add debug info for emulator

* remove sdk list to check perf

* Use host GPU accel

* Revert "remove sdk list to check perf"

This reverts commit ad495915406eba478c6ace56af992b3f7e6a9363.

* Install correct NDK version

* fix ndk install

* record videos and logs

* Fix ndk unzip

* fix script

* fix npm install

* update ndk version

* Try getting NDK from sdkmanager update

* change gpu

* always run artifact (video) upload

* save emulator logs

* create artifacts folder

* Run tests without google apis

* Re-work apk naming and signing to work with detox

* Skip caching - it did not save time overall

* Run CI tests on release build

* fallback to debug signingconfig if release keystore is not availabel

* fix Bitrise builds

* Fix beta-release workflow

* Fix deploy-release workflow

* Ensure npm version name is set

gmaclennan committed:
feat: 📏 Add scale bar to map & improve battery usage (#356)

gmaclennan committed:
chore: Fix s3 upload naming

gmaclennan committed:
chore: Fix debug applicationId

gmaclennan committed:
chore(release): 2.1.0-beta.0

gmaclennan committed:
chore: Merge CHANGELOG from 2.0.0-beta releases

gmaclennan committed:
fix: Fix sync errors by locking discovery swarm to v6.0.0

gmaclennan committed:
chore: Also run Github CI on pull requests

gmaclennan committed:
chore: Version naming for new release process

gmaclennan committed:
chore(release): prepare for 2.1.0

gmaclennan committed:
chore: typo in docs

gmaclennan committed:
chore: Adjust when CI runs

PRs on Bitrise do not have access to secret variables, so we do not run the release candidate workflow based on PR, instead on the branch push.

gmaclennan committed:
chore: Fix e2e tests

okdistribute commented:
Great work! What's RC.760 stand for?

gmaclennan commented:
“RC” is for “Release Candidate”. As for the number “760”, I wanted to automate builds on each push to the PR branch, and I wanted an incrementing number so we know what the latest release candidate is, so 760 is the build number (a number that increments every time CI runs). We also use the build number as the “version code” of the app - every android app needs an integer version code and it must always increment - you cannot downgrade to a lower version code ever. When I finish the GitHub actions, every push to a release PR will leave a comment with a link to a new release candidate build. For now I need to do that manually if QA finds any bugs with this RC.

gmaclennan committed:
fix: Move attribution (i) to bottom-right (avoid overlap with scale bar)

gmaclennan committed:
fix: Fix crash during sync progress (updates progress circle library)

gmaclennan commented:
New Release Candidate build 762 for testing. Fixes:

* Move attribution (i) to bottom-right (avoid overlap with scale bar)
* Fix crash during sync progress (updates progress circle library)

**APKs for testing**
- [QA v2.1.0-RC.762](https://mapeo-apks.s3.amazonaws.com/beta-testing/v2.1.0/mapeo-v2.1.0-RC.762-QA-debug.apk): You can install this alongside your existing Mapeo App and test out new features without affecting your existing data. Look for the red logo.
- ⚠️[v2.1.0-RC.762 APK](https://mapeo-apks.s3.amazonaws.com/beta-testing/v2.1.0/mapeo-v2.1.0-RC.762-universal.apk): This will replace your existing Mapeo installation and you will not be able to downgrade - if it does not work you will need to wait for a fix.

aliya-ryan commented:
You have asked me to review this PR but I don't know what review means. Is that to do with testing something - obvs. you don't want me to look at the code but happy to do whatever it is I can.

gmaclennan commented:
Hi @aliya-ryan sorry for the confusion, I'm still figuring out the details of how to improve this release process and document it all.

The idea of assigning you and @jencastrodoesstuff as reviewers of this PR is to have a mechanism for recording sign-off on the new release. What this means for you and Jen is to test the Mapeo Release Candidate, which is linked in the [issue above](https://github.com/digidem/mapeo-mobile/pull/358#issuecomment-610298413). Once you have tested it you can add your review by clicking "Add Review" at the top of this page. If it the app works well and you think it is ready for release, you can select "Approve" and then "Submit review". If you think something is broken and needs fixed, then you can select "Request changes" and you can add a description of what is not working, and submit your review.

Merging this PR will trigger this version of Mapeo Mobile (v2.1.0) to be released†, which means that it will become available to download on our website and on the Google Play Store. But this PR cannot be merged until both you and Jen have approved it.

This way we can have a clear sign-off process on releases, and the process is clearly communicated and anybody can click "Squash and Merge" to actually create the release once all the reviews are approved (e.g. reduced dependency on me).

So, in summary what I am requesting from you is to download the Mapeo build linked above, test it works ok, and approve this PR if it does. Please let me know how this process works for you.

†This isn't actually true yet, I haven't automated all steps in the process yet.

jencastrodoesstuff commented:
@gmaclennan I appreciate the work going into reducing dependency on you. 
Long term this will surely help. I very much understand that building the structures (this release page) and asking that we actively participate in the process to prevent releases with untested issues going into the world makes sense. But this is all extremely new. 
We have been referencing https://www.notion.so/digidem/Release-Process-REQUIRES-REVISION-353c3c0d97ec47078c3e6218bccb67e0 for the process and already noted that things need updating... It feels like we are testing a testing process which is ok. but it is very disorienting to go through this without having an idea of what high level goals and checkboxes we are addressing.
Testing scripts are being updated slowly. I can focus on that for the next while as I execute testing on Mm & Md. Knowing if they are actually useful in identifying the things the tech team needs to know is the second step for me. The where and how to put them may be here. but is the issue resolution process really best here? If approval from me may in the future initiate a build for release to be made then our testing scripts need to be thorough and reviewed by the tech team. 
We are trying to eliminate human errors in technology as humans with technology... Tricky.

gmaclennan committed:
feat: Device name remains the same across app restarts (#369)

aliya-ryan commented:
Test of release on Samsung Galaxy Prime+ and Ulefone Armor 3T

The GPS button in Mapeo did not display correctly on either phone initially. 
On the Ulefone the first time I opened it, the button remained in searching mode, even when it had found GPS coordinates. Once I pressed + to add a point, from then on it displayed the accuracy properly. On the Samsung it didn't change from 'searching' even when it had a GPS coordinates as evident on the GPS details screen and on adding a point, even after taking lots of points. There was no difficulty adding a point with it on 'searching' mode, and the point then appeared in the right place, and the blue dot was in the right place, but the 'searching' was confusing and misleading.
- [ ]  Much later when I was testing something else on the Samsung, I turned the GPS location button on and off on the phone, and after that it did display the accuracy correctly within Mapeo

aliya-ryan commented:
I deleted the maptiles off phone to see what would happen. With no maptiles installed, and original mapbox map folder deleted. It has a white screen with "error loading map", and none of the points appear. Can still seem to take points, but the map face remains totally white and the scale does not change if pinch. GPS button works correctly without map.
As this is a very likely thing that might happen for partners, it would be good if a screen appeared telling them what was going on and that they could still collect data - would also be good if the points could still appear even if no map behind them (no way of getting a very very undetailed map in as default so that people could see they were at least in the right country??).

aliya-ryan commented:
If the GPS of the phone is switched off and then you try and take a point, in the place where the coordinates would normally appear it says "buscando" where it would be better it would say "Sin GPS" like the GPS button does, or GPS device switched off, or something.

aliya-ryan commented:
The testing script says there should be a shutter sound when you take a photo. I can't remember if I've heard one before or not. But neither phone had a shutter sound when photo was taken.

aliya-ryan commented:
If a photo was taken landscape on the Ulefone then it did not always turn it to landscape as expected. Usually it did, but on a couple of occasions it didn't. It may be that I righted the phone very quickly after taking photo, but photo looks fine, just wrong way round.
Might we be able to add a "turn photo" button anywhere at some point?

aliya-ryan commented:
Sharing:
- I tried to share with gmail on a phone where it didn't already have my account set up. Once I realised and try to go back this didn't work, and I had to quit app and reopen to get back to Mapeo.
- Sharing on a phone where I did have gmail set up worked as I would expect.
- Sharing via bluetooth to my computer from both phones was unsuccessful.
- Sharing via whatsapp works, but if there is more than one photo then it sends one message with all info for each photo.
- I shared with google maps - it only seems to recognise the photo info and you have to them tell it where to place it (on google maps). I don't know if it worked (photo is not there but perhaps it gets reviewed).

aliya-ryan commented:
Details screen - dot does not appear on map in expected place - map is there and the pointer points to right place but I would expect an orange dot above the pointer.

aliya-ryan commented:
Collecting details
- If there are lots of questions in the list then you have to go through all of them before you can save. Would be good if there was a shortcut to save so that you coudl get out of it quickly.

aliya-ryan commented:
Photo delete
I thought - but perhaps i remember wrongly - that you used to be able to delete photographs after taking them, but this isn't possible now. Maybe it never was.

aliya-ryan commented:
Editing observations
If you edit an observation, and then press the go back button rather than the save button, the message says "discard observation" / descartar... when really it should say discard edits as you do not delete the whole observation by not saving changes.

aliya-ryan commented:
Details 
The little blue lightening bolt that appears in top right of screen in details of points taken on another phone. I understand what it means but I don't think it's obvious - maybe it could be a button and if pressed it explained why you can't edit that point.

aliya-ryan commented:
Aliya tested Mm release. Having problems with the settings loader, and also with the sync to computer. But everything else tested and any issues I found reported above.

aliya-ryan commented:
Settings loader does not work on either phone - it takes me to a screen allowing me to choose the file, but when i choose it simply returns to the old screen showing old configurations. If I load settings the old way by copying in folder it works.

aliya-ryan commented:
Project Key does not load correctly.
On both the configurations detail screen and on the sync screen the Project Key is listed as MAPEO. This does not change if you change the configurations, and I know it is not correct for the configurations currently within Mapeo.
Perhaps this is why I can't get it to be recognised on Mapeo Desktop?

aliya-ryan commented:
Mm not picked up by MD on sync

Tried multiple times, quitting Mapeo D and Mm, trying on different networks (hotspot and TP link), trying with different versions of the settings installed, but I can not get the phone and computer to pick each other up. Using Mapeo 5.1.0 on desktop - I had also tried with Mapeo 5.0.0 with similar failure.

My only guess currently is that this is to do with the project key not being processed correctly on the phones and hence not matching the one on the computer (see previous comment).

gmaclennan committed:
fix: Fix photos rotated incorrectly when the phone is rotated soon after photo capture

Use the phone accelerometer data at the moment the photo is taken rather
than after capture

okdistribute commented:
Documenting a bug reported by Jen today, with Md 5.1.0 -- could be a bug in Mapeo Core, although it doesn't seem entirely critical, it is pretty strange. https://github.com/digidem/mapeo-desktop/issues/328

okdistribute commented:
Found another sync bug that may not be new with this version, but I figured I'd report:

1. Take observations with many photos on phone, enough so sync will take at least a few seconds with progress
2. Open desktop sync
3. Open phone sync
4. Start sync 
5. Close desktop app
6. See an error (expected):
![Screenshot_20200417-144915_Mapeo QA](https://user-images.githubusercontent.com/633012/79617229-d2acc880-80bb-11ea-9a57-cb97b426529f.jpg)

7. Open Mapeo Desktop again, try to syncronize

Mapeo gets stuck in this state on the phone, but the Desktop app seems to continue just fine and finishes. You can see the photos and everything looks good on the Desktop
![Screenshot_20200417-145041_Mapeo QA](https://user-images.githubusercontent.com/633012/79617267-eb1ce300-80bb-11ea-89f3-33f4adccb148.jpg)

9. Close the sync screen on mobile and try to find the Desktop app again --> the mobile app is stuck in a 'searching...' state. I'm not able to get them to find eachother again until I "force quit" the application on the phone and restart it. Then they are able to find each other.

gmaclennan commented:
@aliya-ryan thanks for the detailed testing. Several of these bugs also exist in previous releases and I suggest we go ahead with this release without resolving them, and work to fix them in the v2.2 release or later. I have created issues for each of them:

- Add option to delete photos from an observation #384 (this has never existed)
- Allow saving an observation without navigating all questions #383
- Show explanation of synced observations #382 
- Cancelling an edit should show "discard edits" #381 
- Sharing an observation does not work consistently #380 
- Include a default offline map #379
- GPS pill in header does not update correctly #378

I think the issues that do need to be resolved for this release are:

- [x] Taking a photo landscape sometimes results in photo being portrait (I think I fixed this and we can check in the next release candidate)
- [ ] Settings loader does not work correctly
- [ ] Project key is not updating after changing settings

The sync errors reported by @jencastrodoesstuff / @okdistribute I have seen before, and I think it's a bunch more work to fix. I suggest moving forwards with this release without fixing this, and work on a fix in the next release.

> Details screen - dot does not appear on map in expected place - map is there and the pointer points to right place but I would expect an orange dot above the pointer.

@aliya-ryan Is what you see different than this?

![screen](https://user-images.githubusercontent.com/290457/79769339-46550c80-832c-11ea-83c6-117d30b8f616.png)

> The testing script says there should be a shutter sound when you take a photo. I can't remember if I've heard one before or not. But neither phone had a shutter sound when photo was taken.

The shutter sound is hard-coded into some phones. This is because of the horrible issue of people taking "upskirt" photos in public, and as a result some governments require that phones always make a shutter sound when taking a photo. This means that some phones purchased in certain countries will have a shutter sound, and others will not. We could add our own shutter sound in Mapeo but currently we do not.

aliya-ryan commented:
Thanks for responses @gmaclennan Gregor,

In relation to the question about details screen. On neither phone did I see the orange dot or the little marker. Just the bubble with UTM coordinates underneath. As in images below. 

I also get this error message repeatedly coming up on both phones. 

![IMG-8556](https://user-images.githubusercontent.com/21277147/79771399-a2b52d80-8326-11ea-86e2-4a3a59805b1b.JPG)
![IMG-8555](https://user-images.githubusercontent.com/21277147/79771425-a8ab0e80-8326-11ea-9134-91a0ce568e6b.JPG)
![IMG-8554](https://user-images.githubusercontent.com/21277147/79771440-acd72c00-8326-11ea-96bd-520cf7b031ae.JPG)

jencastrodoesstuff commented:
@aliya-ryan those is a fascinating series of bugs. I'm not experiencing all of them. 

1. Both Samsungs are showing the orange point location icon on the observation screen. 
2. Settings loader - I had some difficulty accessing .mapeosettings files because "internal memory" (from the left side menu) showed it was empty so I had to go to "Downloads" and then run a search for the folder I pushed into the documents folder I named MAPEO-config-files. Once I selected one of those I could switch configurations, config names and project keys updated on the config setting screen, and I could collect new points and review observations with the new config category names. @gmaclennan is this something to do with app permissions to access files? 

Your other observations are consistent with mine. 
For some of the UX enhancements, I have been adding them as pink highlighted bullets called "Possible desired behavior" within the [testing script ](https://www.notion.so/digidem/Script-for-MAPEO-Mobile-QA-UPDATE-IN-PROGRESS-56fa80375bb94740913b002bf07c1609) to be collected and discussed since they are non-critical.  Would it make sense to gather them all there until the next TP call and, flag and report the urgent UX issues here? Would that help the triage process @gmaclennan ?

gmaclennan commented:
@jencastrodoesstuff and @aliya-ryan could you both confirm which of the two linked variants you are testing with? I.e the QA variant (red icon) or the regular variant (dark blue icon)? Trying to figure out if the differences are due to that.

@jencastrodoesstuff regarding other UX issues, I'd like to keep this thread to only urgent issues that might block us releasing this to partners. It contains many fixes which are an improvement on what is in the past release, and I'd like to avoid adding new things (e.g. improving some UX issues) which might introduce additional bugs and delay the release.

Regarding permissions for files, that has been very tricky to get right with Android. Unfortunately Android has made changes to the way apps can access the filesystem of the phone in every release of Android, and to make matters more complicated different manufacturers modify Android in different ways. Could @aliya-ryan and @jencastrodoesstuff report the Android versions that your test phones are running?

gmaclennan committed:
fix: Update to latest Mapbox. Potentially fixes missing map icon.

aliya-ryan commented:
@gmaclennan I am using the QA version for the testing that came up with the above series of bugs

On Ulefone it is using Android version 8.1.0

On Samsung is using version 6.0.1

aliya-ryan commented:
@gmaclennan in relation to the map icon dot not appearing - on both the Samsung and the ulefone with the new apk you sent the dots are now appearing as one would expect.

jencastrodoesstuff commented:
didn't see that we are now on 
QA v2.1.0-RC.775

I was testing RC.762
The RC version # seems important. My red QA version of Mm has been on my phones and updating in similar ways to my standard (blue) release versions so the #'s matter to me. 

Using Android versions: 10 & 9

gmaclennan commented:
@jencastrodoesstuff not to worry about it, I just updated it with a couple of small fixes for the bug Aliya was seeing, but didn't see the need to notify you until I have added some more fixes which I'm working on right now.

Can you confirm which you were using (red or blue) when doing the testing?

jencastrodoesstuff commented:
Since this process has been introduced I only test with red

gmaclennan committed:
chore: Better reporting of error message after config update

gmaclennan committed:
fix: Support presets version 3 (only breaks compat with iD, not MM)

gmaclennan committed:
fix(update-config): Don't import presets if version check fails

gmaclennan committed:
fix: Fix changing config with new project key (fix sync not working after change)

Prior to this fix, changing the project key did not change the encryption
key that is used by multifeed, or the discovery key

gmaclennan committed:
fix: Ensure that presets are not cached after config is changed

gmaclennan committed:
chore: Report config import errors to bugsnag

gmaclennan committed:
fix: Show error to user if there is an error importing config

gmaclennan committed:
chore: use mapeoCore.close() and patch in cleanup of hypercores and idx

@mapeo/core v8 does not close the hypercores in the close() method.

gmaclennan committed:
chore: Update KNOWN_ISSUES with bugs that will not be fixed in v2.1

gmaclennan committed:
chore: Fix bugsnag error reporting

okdistribute commented on src/backend/server.js:
```diff
@@ -358,7 +358,15 @@ function createServer({ privateStorage, sharedStorage, flavor }) {
+        // TODO: mapeoCore.close() does not close the underlying hyperlogs or
+        // stop indexing in @mapeo/core v8.1.3. This was added in v9 which we
+        // are not yet using. When we switch to v9 we can remove the extra
+        // cleanup of hypercores and the sync pause here.
```
can we put this in mapeo/core v8 branch for v8.1.4?

gmaclennan replied:
Yes, would support that!

gmaclennan committed:
fix(ObservationView): Fix display of observation icon on inset map

Switch to render a react native Image as the icon instead of rendering
a symbol layer in the map, which was causing some rendering issues on
certain devices. Since the map is static (cannot be panned or zoomed) we
can just render the image in the center of the map and it will be in the
correct location.

gmaclennan committed:
fix(MapView): Improve map performance and device compatibility

Render main map onto a GLSurfaceView, which should improve performance.
See #387

gmaclennan committed:
fix(ObservationView): Speed up initial load of inset map

gmaclennan committed:
chore: Don't re-fetch deviceId (is cached and does not change)

gmaclennan committed:
fix(SyncPeer): Make sync button respond immediately to press

Pressing the button did not immediately do anything - it waited on a peer
update from mapeo core. In some circumstances this does not arrive for
some time (because the sync starts sending first before receiving, so
no peer updates come through). This caused users to press the button
multiple times.

jencastrodoesstuff commented:
— *MAPEO Issue* —
May 5, 2020, 06:15 PM
UTM 18T 607894 5043333

One icon is not updating even though the data folder contents omly has the new version of the icon with the same file name.

_Issue Priority:_
Minor inconvenience
_Issue Reproducibility:_
I'm not sure how
_Issue Context:_
Updating different builds of this config file with changes to the dd-mapeo-bug icon. Using the config importer.
Preset folder was deleted a number of times and the configuration imported a few times. All updated icons changed except for this one
_Expected Behaviour:_
That the updated build with the updated icon would not display the old icon. Just the new one
_Equipment & OS:_
Galaxy  s10e android 10



_Observer:_
Jen
_Affiliation:_
Dd

![5b09679093db42c08177144924c28814](https://user-images.githubusercontent.com/33264095/81124587-d3d75580-8f03-11ea-9258-899b9d689f8a.jpg)
![e3f0ae009c445f9c5d3f27ff15f7f29b](https://user-images.githubusercontent.com/33264095/81124588-d3d75580-8f03-11ea-80cf-621346da0c39.jpg)

— Sent from Mapeo —

jencastrodoesstuff commented:
— *MAPEO Issue* —
May 5, 2020, 06:44 PM
UTM 18T 607887 5043326

Warning screen on Mm

_Issue Type:_
Error Message,Camera issue
_Issue Priority:_
Minor inconvenience
_Issue Reproducibility:_
I'm not sure how
_Issue Context:_
Creating a new observation with new configurations imported last night
After selecting category a warning message appears

_Expected Behaviour:_
No warning, no issues with previously functioning features
_Equipment & OS:_
Galaxy s10e android 10
_Observer:_
Jen
_Affiliation:_
Dd

— Sent from Mapeo — 

![8932af89e7166709150a57c058bb47a7](https://user-images.githubusercontent.com/33264095/81124672-0719e480-8f04-11ea-9451-12bd0aac37f3.jpg)
![204561ddd5957c390a4ea82426cf1c13](https://user-images.githubusercontent.com/33264095/81124673-07b27b00-8f04-11ea-84ab-cdb1d56cc150.jpg)

— Sent from Mapeo —

jencastrodoesstuff commented:
— *MAPEO Issue* —
May 5, 2020, 08:16 PM
UTM 18T 607885 5043332

Config loading not working on older android 7
The mapeosetting package was not a visible option on in the downloads folder where they were transfered successfully from mac android file transfer

The mapeosetting package was not a visible "my files" app either
They do appear in ES File explorer App

Using google drive link the mapeosettings packages were visible via the Mm file picker. After selection the was a greyed text delay and the it jumped back to normal colouring but the was no settings change
Restarted to test again 3 x and still not working.

_Issue Type:_
Configuration loading issue
_Issue Priority:_
Major issue
_Issue Reproducibility:_
Yes
_Issue Context:_
Newly installed Mm v2.1.0 rc 775
Mapeosettings package transfered to download folder of device memory. (No sdcard)
Initial start up normal.

_Expected Behaviour:_
Load config packge
Confirm new configurations
New categories appear after hitting capture
_Equipment & OS:_
Samsung j5 android7
_Observer:_
Jen
_Affiliation:_
Dd

— Sent from Mapeo — 
[(video link)](https://drive.google.com/file/d/15yyXQOLza7S6AvOnegpUvu-puhhiS74G/view?usp=sharing)  - was made for Kelly until I realized that my google drive workaround didn't work either


![4d115cda5d5c7511c80484438b754af4](https://user-images.githubusercontent.com/33264095/81127969-e5255f80-8f0d-11ea-88e1-881890bc5c04.jpg)


![d1847c6d779429a9964faf3904091046](https://user-images.githubusercontent.com/33264095/81127972-e5bdf600-8f0d-11ea-9a6c-383bd5628ee1.jpg)


![4c68746d0530535edd599a089c4d0956](https://user-images.githubusercontent.com/33264095/81127968-e5255f80-8f0d-11ea-8afa-c5a3923427eb.jpg)
![56e32ba1ab2c08d362d57ae85228f32c](https://user-images.githubusercontent.com/33264095/81128160-71378700-8f0e-11ea-9487-9d0f2d881233.jpg)
![16de9a79ed1fc669ececb67d1d8a3069](https://user-images.githubusercontent.com/33264095/81128624-b4dec080-8f0f-11ea-8747-fcbf6cd4e33e.jpg)

gmaclennan committed:
fix: Fix config loading on Samsung devices

Samsung throws an EPERM error if tar-fs tries to set utime on extracted
files

gmaclennan commented:
@jencastrodoesstuff thanks for the detailed reports.

1. Not seeing the file in downloads.

This is weirdness in older versions of Android and _some_ devices. The issue is that the "Downloads" folder that is visible from apps is not the same as the "Downloads" folder that you see from Android File Transfer. I do not know a workaround, other than downloading via an email / google chrome (and it will go to the "correct" downloads folder) then you should be able to see it from the app. I don't know whether putting it in a different folder will work.

2. Presets not actually updating

I managed to debug an issue with samsung phones today, and the fix in the [latest build RC.781](https://mapeo-apks.s3.amazonaws.com/beta-testing/v2.1.0/mapeo-v2.1.0-RC.781-QA-debug.apk) might fix it if your error was caused by the same bug. If not I'd like to troubleshoot directly on your phone over zoom soon.

3. "Yellow box" warnings

This is helpful seeing these. However these do not show on the version we release and they are just warnings "this _might_ be a problem at some point" but they I think they are all ok issues at the moment.

4. New icon not appearing

I would be interested to know if the updated icon did eventually appear. It could be a caching issue. If it has appeared after 24 hours, then I will see if I can add a fix to skip the cache somehow.

jencastrodoesstuff commented:
quick
 1. Damn that's anoying. sd's that aren't sd's, downloads that and not the
same as downloads.  Android must be a nightmare to program
2. Thanks!! want to test this today
3. I'll try to get better at tracking when these appear
4. still not showing. on my s8 the right icon does appear. Very strange.
Ghosts


*Jen Castro*
*(Remote from Montreal)*
*Program Coordinator - **Digital Democracy*
Tel Perú:  +51(1)7007968 , Canada: +16479455252 , WhatsApp: +17789984058,
Skype:  live:jen_5964

**Pido disculpas por cualquier error gramática u ortográfica. Estoy en el
proceso de mejorar mi nivel de castellano y aprecio su paciencia.


On Wed, May 6, 2020 at 12:29 PM Gregor MacLennan <notifications@github.com>
wrote:

> @jencastrodoesstuff <https://github.com/jencastrodoesstuff> thanks for
> the detailed reports.
>
>    1. Not seeing the file in downloads.
>
> This is weirdness in older versions of Android and *some* devices. The
> issue is that the "Downloads" folder that is visible from apps is not the
> same as the "Downloads" folder that you see from Android File Transfer. I
> do not know a workaround, other than downloading via an email / google
> chrome (and it will go to the "correct" downloads folder) then you should
> be able to see it from the app. I don't know whether putting it in a
> different folder will work.
>
>    1. Presets not actually updating
>
> I managed to debug an issue with samsung phones today, and the fix in the latest
> build RC.781
> <https://mapeo-apks.s3.amazonaws.com/beta-testing/v2.1.0/mapeo-v2.1.0-RC.781-QA-debug.apk>
> might fix it if your error was caused by the same bug. If not I'd like to
> troubleshoot directly on your phone over zoom soon.
>
>    1. "Yellow box" warnings
>
> This is helpful seeing these. However these do not show on the version we
> release and they are just warnings "this *might* be a problem at some
> point" but they I think they are all ok issues at the moment.
>
>    1. New icon not appearing
>
> I would be interested to know if the updated icon did eventually appear.
> It could be a caching issue. If it has appeared after 24 hours, then I will
> see if I can add a fix to skip the cache somehow.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/digidem/mapeo-mobile/pull/358#issuecomment-624752483>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH5ZDX3LQN3VTJW2MJUME3LRQGF45ANCNFSM4LVAV37Q>
> .
>

okdistribute commented on src/backend/server.js:
```diff
@@ -217,5 +321,121 @@ function createServer({ privateStorage, sharedStorage }) {
+    // the discovery swarm, and close all the random-access-storage instances
+    // and the leveldb instance
+    const close = function close(cb) {
+      let pending = storages.length + 1;
```
would this be storages.length + 2?

one for indexDb and the other for osm.core._logs.close?

gmaclennan replied:
Thanks for the catch @okdistribute! I forgot to modify this when I added _logs.close. I've started to lean towards async/await for a lot of things, bugs like this and unhandled errors are too easy to miss with callback patterns.

gmaclennan commented on src/backend/server.js:
```diff
@@ -217,5 +321,121 @@ function createServer({ privateStorage, sharedStorage }) {
+    // the discovery swarm, and close all the random-access-storage instances
+    // and the leveldb instance
+    const close = function close(cb) {
+      let pending = storages.length + 1;
```
```suggestion
      let pending = storages.length + 2;
```

gmaclennan committed:
Fix close function to wait for everything to close

gmaclennan committed:
Remove unnecessary close of hypercore storage

Multifeed.close closes hypercore, which closes all the storage

gmaclennan committed:
chore: Rename pending release to v5 and update changelog

gmaclennan committed:
feat: Add Brazilian Portuguese translations (#412)

okdistribute committed:
chore(deps): update to latest mapeo core

okdistribute committed:
abstract usePeers from desktop

okdistribute committed:
fix: pull in code from desktop for usePeers state manager

gmaclennan committed:
chore: Add missing intl messages

okdistribute committed:
fix: get sync to render properly with new mapeo core

okdistribute committed:
chore(deps): update to latest mapeo core

okdistribute committed:
fix: sync on peer device should show progress events on local device

okdistribute committed:
fix: pass through 'connected' state to peer list

okdistribute committed:
fix: show indeterminate state when progress is 1

okdistribute commented:
Bug Report: https://github.com/digidem/mapeo-mobile/issues/415

gmaclennan commented:
@aliya-ryan @jencastrodoesstuff @okdistribute Do you think bug #415 (can't sync after peer disconnects and re-connects) should block this release? I.e. can we release with this bug, or should we try to fix it first?

gmaclennan committed:
fix: Fix Portuguese (Brazilian) translations

okdistribute committed:
fix: show a yellow icon when disconnected

okdistribute committed:
fix: When WiFi connection state changes, refresh peer list

okdistribute committed:
fix: Ignore errors (e.g., ECONNABORTED) that are now caught by the
'connected' property

okdistribute committed:
fix: 'missing data' error will not cancel large syncs too early

aliya-ryan commented:
**Test results** 
_Samsung Galaxy Grand Prime+
Android v 6.0.1_
Full notes in script: https://www.notion.so/digidem/9-July-2020-mapeo-v5-0-0-RC-810-QA-debug-apk-0474cf2d693d4a4cb4da2c7fa7c89c1d

I just did minimal testing of most essential parts - according to shortened script created by @jencastrodoesstuff  for Tom R. to test.

I found no errors I would consider to be blocking (but did not do full test)

I tested sync with my mac, using v5.3.0-beta.2
I could not test sync with another phone as gave my second phone to Tom R. to use for testing

**Comments:**
UI
- When start up the QA, it appears with the blue background and Mapeo logo, would make more sense if the QA could start up with red background instead, to help differentiate.
- The blue dot on the map screen showing current location appears underneath the orange dots of points - I think it would make more sense the other way round
- When I started up offline, it said 'error loading map' - perhaps more useful to say something more instructional like "error loading map, - try going online or load an offline map"
- In the language chooser it might make sense if the current chosen language appears at top of list, perhaps with font/colour differentiation or something.
- When go to sync screen without any wifi network - the Wifi says: Wifi: <unknown ssid> - this isn't very useful, could it not say "device not connected to a wifi network" or "connect to a wifi network to syncronise" or something like that.
- The script says there should be an optoin to "open settings" to go to the wifi menu - there is no "open settings" button - maybe the UI has changed @jencastrodoesstuff do we need to update the script?

Bugs / Issues
- When I opened Mm, it was not connected to internet. The GPS details button said 'searching' but if clicked on it then the points had already been captured. Stayed like this for ages, until I connected to internet - could this be something to do with there not being a map? I could not reproduce this, despite opening and closing Mapeo a few times. - not blocking
- the online map didn't load once I connected to internet until I had quit and reopened Mapeo - not blocking
- When syncing with computer, for a brief moment on the phone it came up with an error message saying something like "Error in Mapeo database" - but it was too fast to take note of and then it said sync complete, and seemed to be complete. I could not reproduce this. - not blocking
- After the sync there was an error message at bottom of screen, see photo below. I dismissed it. 
![IMG_8897](https://user-images.githubusercontent.com/21277147/87036243-6b964000-c1e2-11ea-81a5-5cba2b876e5b.JPG) - not blocking

okdistribute committed:
fix: Improve the formatting of Mapeo Alerts fixes #401

okdistribute committed:
chore: for flow, ensure fields and preset are always present

okdistribute committed:
fix: some last fixes for mapeo alerts

okdistribute committed:
chore: ensure flow understands we use select_multiple now too

jencastrodoesstuff commented:
Gave RC830 a full test including install, load configurations, add, edit, share, delete, sync with Md &Mm.   Karissa did minor corrections to characters for share function that she sent me. 
There is that strange issue of data not appearing on the map for a while even though it is in the observation list. Tom identified it on his summary as well. This is a strange behavior I encountered with a guardaparque in the RCA but we couldn't figure out if it was time that made the points appear or connectivity to the internet.  Can figure out the sequence for testing this for the next release.

jencastrodoesstuff commented:
What is the process for us to review apps that are ready to release?

okdistribute committed:
fix: ignore missing data errors and errors that are stale

okdistribute committed:
chore(deps): bump to @mapeo/core v8.3.2 for missing data fixes

gmaclennan committed:
chore: Update package-lock to v5.0.0

gmaclennan committed:
chore: Update to mapeo-default-settings@3.0.0

okdistribute committed:
fix: Ensure sync errors are always propagated to frontend.

gmaclennan committed:
fix: Fix observations not rendering when GPS is off (#431)

Previously, <MapboxGL.Camera> was not rendered when location services
were off. This caused the <MapboxGL.ShapeSource> to not render (for
unknown reasons). This fix always renders MapboxGL.Camera and ignores
location updates if the map is not in follow mode.

gmaclennan committed:
fix: Update Portuguese translations

* New translations en.json (French)

* New translations en.json (Spanish)

* New translations en.json (Vietnamese)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Indonesian)

* New translations en.json (Khmer)

* New translations en.json (Thai)

* New translations en.json (Hindi)

* New translations en.json (Malay)

* New translations en.json (Burmese)

* New translations en.json (Nepali)

* New translations en.json (Lao)

gmaclennan committed:
fix: Update Thai and Nepali translations

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Nepali)

* New translations en.json (Nepali)

gmaclennan committed:
chore: Update changelog (prepare for release)

gmaclennan committed:
chore: Update KNOWN_ISSUES for v5.0.0 release

jencastrodoesstuff reviewed (approved):
**Tested and passed***
Install
load config
online map loads
_*known issue: points don't appear in mapview when there is no online map and no offline map_
offline map loads
center to location
live track location on map
GPS, precision update and details screen
observation points appear on map with GPS OFF

new observation: select category, add description add photos, add details, save
edit observation: edit category, edit description, add photos, edit details, save
_*known issue: photos requires datestamp for verification purposes_
 manual entry GPS coordinates

Share with WhatsApp - custom formatting
Share with Email - custom subject line

Delete

Sync with Wifi
Syncing big databases from desktop* and mobile (900+ observations)
_*known issue: sync process termination and error front end issues_
Resyncing after errors (from previously interrupted syncs)

**PASS**
_Release the hounds!!_

gmaclennan committed:
chore: Roll back to react-native-mapbox-gl/maps@8.0.0

The 8.1.0-rc.2 seemed to be the cause of errors fetching tiles when the
device is offline

gmaclennan committed:
fix: Turn off dev mode (yellow warning boxes) in universal build
